elfcore/src/coredump.rs

```diff
@@ -917,7 +921,11 @@ fn process_vm_readv_works() -> bool {
 /// * `writer` - a `std::io::Write` the data is sent to.
 /// * `pv` - a `ProcessView` reference.
 ///
-pub fn write_core_dump<T: Write>(writer: T, pv: &ProcessView) -> Result<usize, CoreError> {
+pub fn write_core_dump<T: Write>(
+    writer: T,
+    pv: &ProcessView,
+    custom_notes: Option<&mut [CustomFileNote]>,
```

Comment:
If we're going to start adding stuff to this (previously-simple) function, I'd suggest that we leave this alone and make a builder pattern.

Also, this is a breaking change.

Reply:
making a builder pattern would also have to be a breaking change, right? If so, do you think I should try to do this in a way that wouldn't be? Originally, I had the custom notes in the processview struct which made it sort of builder like, but I changed it to this way to avoid having make a vec to own CustomFileNotes structs, but I suppose it probably isn't a big deal.

Reply:
The idea is that you'd keep the existing `write_core_dump` function as is, but internally it would call some new `CoreDumpBuilder::new().write(writer, pv)` (or maybe `pv.core_dump().write(writer)` would be a nicer way to write this).

`CoreDumpBuilder` would then have some method `add_custom_note` for this new use case. So people who need the new functionality would switch to the builder, but existing code could continue to use the "legacy" function.

Reply:
Ok I'll see what I can do

Reply:
I'll see if I can test the new functionality in the example program so I can get rid of the #[allow(dead_code)], and still trying to think of a way to have the backwards compatibility without cloning the process view, but here is a first attempt (I don't think I've written a builder pattern before).

Reply:
Steven, Chris, and I are thinking that its not worth it to avoid a breaking change. We can remove the old API and call this 1.2. I don't think we need to expose the ProcessView struct to the user.

Reply:
This would have to be 2.0. 1.2 can't have breaking changes.

I think this is a bad reason to rev the major version.

Reply:
(I don't know why we declared 1.0 so soon, but there you go.)

Reply:
Ok there shouldn't be any breaking changes now, and the new functionality uses a builder